LibTest/ffi/UintPtr/UintPtr_A01_t01.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The C uintptr_t type.
+///
+/// The [UintPtr] type is a native type, and should not be constructed in Dart
+/// code. It occurs only in native type signatures and as annotation on [Struct]
+/// and [Union] fields.
+///
+/// @description Checks that this represents C uintptr_t type
+/// @author sgrekhov22@gmail.com
+
+import "dart:ffi";
+import "package:ffi/ffi.dart";
+import "../../../Utils/expect.dart";
+
+void main() {
+  Pointer<UintPtr> p1 = calloc<UintPtr>();
+  try {
+    p1.value = 42;
+    Expect.equals(42, p1.value);
+    p1.value = -42;
+    Expect.equals(-42, p1.value);
```

Comment:
@dcharkes is it Ok, that `UintPtr` returns a negative value?

Reply:
I think this is because the underlying representation on x64 is Uint64, in which the upper half of numbers is exposed as dart negative integers.

However, on Uint32 this will not happen, so this test will fail on 32 bit platforms.

So, we should rewrite the test to either accept both outcomes (0x100000000-42 on 32 platforms) or remove the test because we have this behavior covered on the Uint64 and Uint32 tests already.